scripts/release_scripts/update_configs_test.py

```diff
@@ -309,94 +313,130 @@ def mock_update_analytics_constants_based_on_config(
-            f.write(constants_text)
-        with self.assertRaisesRegex(
-            Exception, 'Error: No SITE_NAME_FOR_ANALYTICS key found.'
-        ):
-            update_configs.update_analytics_constants_based_on_config(
-                temp_analytics_constants_config_path,
-                temp_constants_path
-            )
-
-        # Testing invalid CAN_SEND_ANALYTICS_EVENTS key.
-        constants_text = (
-            '  "GA_ANALYTICS_ID": "123"\n'
-            '  "SITE_NAME_FOR_ANALYTICS": "site-name"\n'
-            '  "can_SEND_analytics_EVENTS": true\n'
-        )
-        with utils.open_file(temp_constants_path, 'w') as f:
-            f.write(constants_text)
-        with self.assertRaisesRegex(
-            Exception, 'Error: No CAN_SEND_ANALYTICS_EVENTS key found.'
-        ):
-            update_configs.update_analytics_constants_based_on_config(
-                temp_analytics_constants_config_path,
-                temp_constants_path
-            )
+        with tempfile.NamedTemporaryFile(mode='w+t') as temp_constants:
+            with tempfile.NamedTemporaryFile(
+                mode='w+t'
+            ) as temp_analytics_constants_config:
+                constants_text = (
+                    '  "GA_ANALYTICS_ID": "123"\n'
+                    '  "SITE_NAME_FOR_ANALYTICS": "site-name"\n'
+                    '  "CAN_SEND_ANALYTICS_EVENTS": true\n'
+                )
+                analytics_constants_config_text = (
+                    '  "GA_ANALYTICS_ID": ""\n'
+                    '  "SITE_NAME_FOR_ANALYTICS": ""\n'
+                    '  "CAN_SEND_ANALYTICS_EVENTS": false\n'
+                )
+                expected_analytics_constants_config_text = (
+                    '  "GA_ANALYTICS_ID": "123"\n'
+                    '  "SITE_NAME_FOR_ANALYTICS": "site-name"\n'
+                    '  "CAN_SEND_ANALYTICS_EVENTS": true\n'
+                )
+                temp_constants.write(constants_text)
+                temp_constants.flush()
+                temp_analytics_constants_config.write(
+                    analytics_constants_config_text)
+                temp_analytics_constants_config.flush()
+
+                update_configs.update_analytics_constants_based_on_config(
+                    temp_analytics_constants_config.name,
+                    temp_constants.name
+                )
+                # We need to reopen the file here because the update method
+                # called above actually creates a new file in the same path
+                # instead of writing in the same file.
+                with utils.open_file(
+                    temp_analytics_constants_config.name, 'r'
+                ) as f:
+                    self.assertEqual(
+                        f.read(),
+                        expected_analytics_constants_config_text
+                    )
+
+    def test_raises_error_with_invalid_ga_analytics_id(self) -> None:
```

Comment:
why are we changing the test functionality?

Reply:
This test case was split into 3. But while resolving merge conflicts I've merged them back together.